omod/src/main/webapp/resources/scripts/directives/select-drug.js

```diff
@@ -16,8 +16,19 @@ angular.module('uicommons.widget.select-drug', [ 'drugService', 'ui.bootstrap' ]
                 $scope.size = attrs.size ? attrs.size : 40;
 
                 $scope.search = function(term) {
-                    return DrugService.getDrugs({ q: term });
-                }
+                    return DrugService.getDrugs({ q: term }).then(function(results) {
+                        // Corresponding html attribute is support-free-text if one wants to support non coded
+                        // the directive would be like <select-drug support-free-text ...other attributes></select-drug>
+                        if(attrs.hasOwnProperty('supportFreeText') && results.length == 0) {
+                            return [{
+                                uuid: null,
+                                display: '"' + term + '"' + ' (NON CODED)',
```

Comment:
I'm thinking it would have been nice to have 'NON CODED' internationalized

Reply:
In addition to the fact that this needs to be i18nized, I'm not sure that "Non coded" is an end-user friendly term. Maybe you should post on Talk asking for ideas about what this should be called?